contracts/operator.sol

```diff
@@ -24,14 +24,27 @@ contract Operators is Claims, ClaimTypes, Bonds {
 
     // STATE UPDATE
 
-    event StateUpdate(address _operator, bytes32 _confirmed_state, uint256 _balance, uint64 _res, uint64 _revision);
+    event StateUpdate(address _operator, bytes32 _confirmed_state, uint64 _revision);
 
     /* TODO: include a bond with this and every claim */
-    function claim_state_update(bytes32 _new_state, uint64 _revision) external {
-        uint64 res = make_claim(digest_claim(msg.sender, ClaimType.STATE_UPDATE, _new_state));
-        emit StateUpdate(msg.sender, _new_state, address(this).balance, res, _revision);
+    function claim_state_update(bytes32 _confirmed_state, uint64 _revision) external {
+        make_claim(digest_claim(msg.sender, ClaimType.STATE_UPDATE, _confirmed_state));
+        emit StateUpdate(msg.sender, _confirmed_state, _revision);
     }
 
+    // This is a hack because we use block depth for confirmation.
```

Comment:
Using block depth for confirmation is not "a hack", it's the natural, cheap and efficient way to manage claims.

Why do we even need a null event? I don't know of any use for it. If for some reason you needed a state update, you could reissue the same one, and/or add some administrative transaction, acknowledgement of previous state update, etc.

Reply:
I agree that using block confirmation is not a hack.
The problem is the following: When running the system on our testnet on laptop, we are the only one making events on the chain. This implies that the block depth does not increase. And so transaction never gets confirmed.
The "NullOperation" is something like your "administrative transaction". We cannot reissue the same transaction because of the require statement that requires a claim not to have been put before.

Reply:
These deserve an issue of their own... https://github.com/AlacrisIO/legicash-facts/issues/27